bluephos/bluephos_pipeline.py

```diff
@@ -78,15 +93,68 @@ def ligand_smiles_reader_generator(ligand_smiles):
         yield row.to_frame().transpose()
 
 
-def get_generator(ligand_smiles, halides, acids, input_dir, t_nn, t_ste):
+def get_generator(ligand_smiles, halides, acids, input_dir, t_nn, t_ste, t_dft):
     """
     Get the appropriate generator based on the input directory presence.
     """
     if ligand_smiles:
         return lambda: ligand_smiles_reader_generator(ligand_smiles)
     elif not input_dir:
         return lambda: ligand_pair_generator(halides, acids)
-    return lambda: rerun_candidate_generator(input_dir, t_nn, t_ste)
+    return lambda: rerun_candidate_generator(input_dir, t_nn, t_ste, t_dft)
+
+
+def build_pipeline_graph(input_dir: str, ligand_smiles: str):
+    """
+    Construct the pipeline graph based on input conditions.
+
+    Args:
+        input_dir (str): Directory containing input parquet files.
+        ligand_smiles (str): Path to the ligand SMILES CSV file.
+
+    Returns:
+        list: A list of task tuples representing the pipeline graph.
+    """
+    # Case 1: Input as halides and acids CSV files
+    if not (input_dir or ligand_smiles):
```

Comment:
The cases here look really similar save for where the pipeline starts. I wonder if you can initialize the full edge list and then pick from there, like:

```py
full_pipeline = [
  ...
]

if ligand_smiles:
  return full_pipeline[1:]  # from NN
if input_dir:
  return full_pipeline[2:]  # from Smiles
return full_pipeline
```